BitbucketV2Proxy.cs

```diff
@@ -324,43 +340,110 @@ public string GetOAuthUri(string state = null, string redirectUri = null)
+                        {
+                            return string.Equals(serviceUri.Host, configUri.Host, StringComparison.OrdinalIgnoreCase);
+                        }
+                    }
+
+                    return false;
+                });
+            }
+        }
+
+        /// <summary>
+        /// Remove Webhook service from v1 API
+        /// </summary>
+        public async Task<bool> RemoveService(string repoUrl, string accessToken, string serviceUrl)
+        {
+            CommonUtils.ValidateNullArgument("repoUrl", repoUrl);
+            CommonUtils.ValidateNullArgument("accessToken", accessToken);
+            CommonUtils.ValidateNullArgument("serviceUrl", serviceUrl);
+
+            var service = await GetService(repoUrl, accessToken, serviceUrl);
+            if (service != null)
+            {
+                var requestUri = BitbucketProxyHelper.GetRequestUri(APiV1BaseUrl, repoUrl, "services", service.id);
+                using (var client = CreateHttpClient(accessToken))
+                using (HttpResponseMessage response = await client.DeleteAsync(requestUri))
+                {
+                    if (!response.IsSuccessStatusCode)
+                    {
+                        throw CreateOAuthException("RemoveService", await response.Content.ReadAsStringAsync(), response.StatusCode);
+                    }
+                }
+            }
+
+            return service != null;
+        }
+
+        private async Task<List<T>> ListPagingItems<T>(HttpClient client, string initRequestUri, string operationName)
+        {
+            // For security and performence purpose, only read the first 30 pages
+            int count = 30;
+            var results = new List<T>();
+            while (!string.IsNullOrWhiteSpace(initRequestUri) && count-- > 0)
             {
-                using (var response = await client.GetAsync(requestUri))
+                using (HttpResponseMessage response = await client.GetAsync(initRequestUri))
                 {
-                    var sshKeys = await this.ProcessResponse<BitbucketProxy.BitbucketSSHKeyFullInfo[]>("GetSSHKey", response);
-                    return sshKeys.FirstOrDefault(info => BitbucketProxy.SSHKeyEquals(info.key, sshKey));
+                    var pagingResult = await ProcessResponse<BitbucketV2Paging<T>>(operationName, response);
+                    results.AddRange(pagingResult.values);
+                    initRequestUri = pagingResult.next;
                 }
             }
+
+            return results;
         }
 
-        private async Task<BitbucketProxy.BitbucketHookInfo> GetWebHookInfo(string repoUrl, string accessToken, string hookUrl)
+        private async Task<BitbucketProxy.BitbucketSSHKeyFullInfo> GetSSHKey(string repoUrl, string accessToken, string sshKey)
+        {
+            var requestUri = BitbucketProxyHelper.GetRequestUri(APiV1BaseUrl, repoUrl, "deploy-keys");
```

Comment:
Created below bug

[https://bitbucket.org/site/master/issues/12746/missing-v2-api-to-add-remove-deployment](https://bitbucket.org/site/master/issues/12746/missing-v2-api-to-add-remove-deployment)

---
In reply to: [64789996](https://github.com/suwatch/AzureSourceControls/pull/4#discussion_r64789996) [](ancestors = 64789996,64696670)